modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/blockstream/BlockStreamSubscription.kt

```diff
@@ -0,0 +1,123 @@
+package jp.co.soramitsu.iroha2.client.blockstream
+
+import io.ktor.client.plugins.websocket.webSocket
+import io.ktor.websocket.close
+import io.ktor.websocket.readBytes
+import jp.co.soramitsu.iroha2.IrohaSdkException
+import jp.co.soramitsu.iroha2.SingletonHolder
+import jp.co.soramitsu.iroha2.cast
+import jp.co.soramitsu.iroha2.client.Iroha2Client
+import jp.co.soramitsu.iroha2.generated.BlockSubscriptionRequest
+import jp.co.soramitsu.iroha2.generated.VersionedBlockMessage
+import jp.co.soramitsu.iroha2.generated.VersionedBlockSubscriptionRequest
+import jp.co.soramitsu.iroha2.toFrame
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import org.slf4j.LoggerFactory
+import java.math.BigInteger
+import java.util.UUID
+import kotlin.coroutines.CoroutineContext
+
+class BlockStreamSubscription private constructor(private val context: BlockStreamContext) : CoroutineScope {
+
+    override val coroutineContext: CoroutineContext = Dispatchers.IO + SupervisorJob()
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    private val source: MutableMap<UUID, BlockStreamStorage> = mutableMapOf()
+    private var initialStorageId: UUID? = null
```

Comment:
why do we need this? I guess it'd be better not to have the priviledged storage

Reply:
see `subscribe` function

Reply:
in `subscribe` function we need it to not calling `expand` few times